components/theme/layout/toc-clerk.tsx

```diff
@@ -0,0 +1,161 @@
+"use client";
```

Comment:
everything hereafter are mostly "ejected" components, with small tweaks to add the features above